src/parse.js

```diff
@@ -260,12 +310,26 @@ function parse(source, root, options) {
         isProto3 = syntax === "proto3";
 
         /* istanbul ignore if */
-        if (!isProto3 && syntax !== "proto2")
+        if (!isProto3 && syntax !== "proto2" || edition)
             throw illegal(syntax, "syntax");
 
         skip(";");
     }
 
+    function parseEdition() {
+        skip("=");
+        edition = readString();
+        isProto3 = true;
+
+        /* istanbul ignore if */
+        if (syntax)
+            throw illegal(syntax, "edition");
+
+        syntax = "proto3";
+        isProto3 = true;
```

Comment:
This seems strange.  Why treat editions files like proto3?

Reply:
Only added the fallback to produce valid schemas that can be tested, including during resolve. I guess another option is to test via the parser only, and for now produce an "editions are not supported" error during resolve?

Reply:
Ah I see, this is just a temporary test hook?  That makes sense, but I definitely like the failure approach better.  That way nobody starts depending on this behavior